theano/gof/cc.py

```diff
@@ -476,11 +476,12 @@ def apply_policy(policy, r, name, sub):
         C{policy[0](r) + policy[1](r) + ...}.
 
     """
-    if isinstance(policy, (list, tuple)):
-        ret = ""
-        for sub_policy in policy:
-            ret += sub_policy(r, name, sub)
-        return ret
+    if not isinstance(policy, (list, tuple)):
+        policy = [policy]
+    ret = ""
+    for sub_policy in policy:
+        ret += sub_policy(r, name, sub)
+    return ret
     return policy(r, name, sub)
```

Comment:
I was wrong here, the old code did the job, I was confused just because I expected `Wrapper.c_init()` to be run, but there are some cases where `c_init()` is skipped and only `c_extract()` is run, e.g. : https://github.com/Theano/Theano/blob/master/theano/gof/cc.py#L686 .

I don't yet understand all the rules here, but it seems that some C codes to init a Theano type should sometimes better be at top of `c_extract()`, instead of in `c_init()`. That's what I have done for Wrapper class,

I'll put back the old code of `apply_policy()`.